### ISSUE TITLE
Player Skill option missing, SC useless without it.

### ISSUE THREAD

**Issue by ubersonic** [simulationcraft/simc#3937]
Version 725.02

Hi, the "player skill" setting appears to have disappeared from the GUI, I need to know where the setting is to re-enable this because I am not a top 1% player so the information SC is currently giving me is useless.

I managed to fix the problem by rolling back to 725.01 but this is obviously not a permanent solution.

Thanks.

*EDIT*

Googling this, it appears to be a deliberate mistake by the designers, I have no idea why you thought making your program useless for 99% of your users was a good idea but this needs fixing ASAP because right now SC is about as useful as maxdps.com.  I don't need to know what DPS my character could do if it's played by a robot that does everything perfectly, I need to know what the DPS is likely to be for a human of my skill level attempting to execute the correct rotation while fighting a boss encounter >.>

**Comment by Mystler** [comment]
I would love an explanation of why that makes SimC *useless*. This is a theorycrafting tool designed to find optimizations concerning gear, rotation and more and the best way to do that is under certain skill assumptions.

I know people like to be humble and reduce the player skill option when they are not top players, however that option was always completely misleading and hence has been removed. The way player skill worked was simply a chance to fall through to another action in the APL. This kind of "fail" was not at all realistic and only skewed simulation results. This was also explained in the other issue you commented on. (https://github.com/simulationcraft/simc/issues/3889)

So, I'm not sure what exactly you expect from SimC. I don't see an issue with simming with proper APL execution and using that for improving your character. If you want to see what DPS you yourself can actually do, the best way is still to go ingame, try and check logs what you did and what you can improve upon.

**Comment by ubersonic** [comment]
Hi,

It makes it useless (to most players) because it can no longer be configured to give an accurate representation of the amount of DPS that can be achieved using a specific set of gear, furthermore because of that it can now even give erroneous results as to which gear/talent/etc choice is optimal.  I.E one gear talent setup may work slightly better for an "elite" player but another may perform slightly better for a "good" player.

Reducing the skill setting in SC has nothing whatsoever to do with being "humble", it's about knowing that you are not in the top 1% of players and that because of that you won't always execute your rotation flawlessly like a robot or press the correct button when several come off cooldown at once and you have fractions of a second to change which key is going down.

I am a "good" player, I am not ashamed of it and I am not being humble about it.  By setting simcraft to "good" I was always able to achieve accurate/realistic results that helped me decide what gear/talent combos to best use, now this ability is lost for me and the majority of players.  We can now only get results for the "elite" players which gives us laughably unrealistic/inaccurate results which may or may not even be correct for us.

TLDR: It was a useful option that for years helped many and hurt none, please bring it back or at least include an option to turn it back on for those who want it.  Because without it this software is now useless to us :(

**Comment by Mystler** [comment]
What I was trying to explain is that it never was a good or accurate option. This "simulation of imperfect rotation" was pretty much random abilities. If results happened to match your DPS than that was more coincidence than anything else.

You keep insisting on that 1% of players number, for some reason. Do you not want to orient yourself on how you *should* (according to current APL research) play? What good are results (still from a perfectly timed robot, btw) that come from random ability usage. What makes you think that player skill option actually "failed" the same way you do? The best approach to player skill simulation would be considering debuff maintenance failure, slowed reaction time, anticipated actions etc., none of which that option ever did.

If you want to argue the accuracy of the results, you should not look at just the DPS value. After all, this is indeed a patchwerk fight with optimal conditions. However arbitrarily reducing your DPS gains you nothing.

> It was a useful option that for years helped many and hurt none [...]

The only way I can see how that option "helped" was by leading to lower, inaccurate results that, idk, make people feel better?

However, if you insist on using that option, you still can and who am I to stop you. ;)

The "player skill" option is called `skill` and is represented by the chance to recognize a condition is actually true. The "good" option had the value `0.95`, so you can just add this to your character profile:

```
skill=0.95
```

Feel free to evaluate how good the robot performs in comparison to you by looking at the sample sequence in the report.

**Comment by ubersonic** [comment]
> This "simulation of imperfect rotation" was pretty much random abilities. If results happened to match your DPS than that was more coincidence than anything else.

Well regardless of what fuzzy logic or whatever it used to achieve it's results, they were very accurate results that allowed for decisions on item/talent choices.  Unlike the new gimped version of SC.


> You keep insisting on that 1% of players number, for some reason.

Because the system is now locked to "elite" player skill, hence giving results that are only accurate/optimat for the most elite of players.


> What good are results (still from a perfectly timed robot, btw) that come from random ability usage.

They were more accurate than the results currently being produced and so were a better information set for making gearing/talent choices.  How the results were achieved by the system isn't really relevant, "if it works, it works".


> The only way I can see how that option "helped" was by leading to lower, inaccurate results that, idk, make people feel better?

It helped in two ways, firstly: the results were more accurate for the average player because they better reflected real world results.  In Example, if I swap out one of my legendaries (sacrificing some useful functionality to gain DPS), the results using the "good" and "elite" skill setting both go up as you would expect, however the result for the "elite" setting go up more (in both raw numbers and as a percentage of the total).  The result is that the change actually looks worth it with the elite setting even though it isn't for a non-elite skilled player.  If I rerun that comparison today using the new version of SC with reduced functionality it will simply give me the impression that I should change items when in reality I should stick.

And not just that but there have historically been points where X talent is better under "elite" setting and Y talent is better under "good" setting, which lead to hilarity when I was out damaging better geared guildmates who had just left SC on default and assumed that talent was right for them.

Secondly: It helped because you often encounter issues where a GM/officer will use SC to sim their raiders then complain if they are not matching the impossible numbers (especially if they think it's related to them doing something wrong gear/spec/etc wise).  This could quickly be fixed in any non world first orientated guild by instructing them to set the PS to "good" to seen realistic results, that is not an option now and so there will be many many players getting wrongly criticised or even gkicked due to this pointless change.


> However, if you insist on using that option, you still can

Thank you for the fix :)

But you really need a toggle in the software for enabling/disabling this functionality, many users wont think to come here and will simply stop using the software if it no longer produces useful data for them.

**Comment by lgkern** [comment]
The problem with the SimC PlayerSkill option is that it is badly designed. The `skill` option is basically a chance that the actor will execute another available action at random instead of the correct one, which isn't really realistic, because for some specs you can break the rotation completely by using a cooldown in an awful time or something similar. You can still use it, but keep in mind that a `skill`, of 95 for instance, will be way worse than a 95th percentile player. As an alternative playing around player latency might yield better results to achieve "real" DPS values, as it emulates SimC having some time to "think" about what to do next.

**Comment by Mystler** [comment]
You quoted me several times and "objected" by saying lower player skill results are accurate. But how do you judge that accuracy? It sounds to me like you just looked at the DPS number. Let me recommend it again: Do a simulation with both player skills, then look at the report -> Action Priority List -> Sample Sequence Table. You will see an example of what the robot does in a fight. Compare the two with each other and with you. This will give you much more of an idea how "accurate" the results really are.

Depending on the spec, i think it is actually easier to perform close to "Elite" than e.g. close to "Average", concerning the ability usage (not DPS).

**Comment by ubersonic** [comment]
> The problem with the SimC PlayerSkill option is that it is badly designed.

I'm not claiming that it's designed well, how it's designed and what it does aren't really relevant to the end user.  The important thing is the results and for the average player the results generated with the "good" preset are significantly more accurate than those generated by the "elite" preset.  As it stands the newest version of SC is churning out less relevant/valid information than the previous version, most players will not be able to achieve the numbers reported and it's not even useful for comparing items/talents/gems/etc because it might not even be correct.

Like I said, I have no issue with the "elite" setting being the default, most players will think to change it.  The issue is it's now hidden in some command line, and this reduction in functionality hurts users.


> It sounds to me like you just looked at the DPS number. Let me recommend it again: Do a simulation with both player skills, then look at the report -> Action Priority List -> Sample Sequence Table. You will see an example of what the robot does in a fight.

It doesn't matter what it does to get the results, what matters is that the results are more accurate in the older version where player skill could be manually set.  And thus they are more useful for making decisions and comparing options, whereas the newer version due to being less accurate/realistic can skew the appearance of results.


*EDIT*

Here's an example of what I'm trying to convey: If I ask my computer to divide 10 by 2, it will multiply it by 0.5 and give me the correct answer, this is because computers are incapable of division.  But Microsoft don't remove division from the calculator on the basis that it's not working it out properly so users don't need the function.

Functionality that many people use has been removed from SC, the program is worse for it.

**Comment by lgkern** [comment]
Well, your example in this case should to be "If I ask my computer to divide 10 by 2, it will multiply it by 0.3 and give me the correct answer but for another question." 
The issue with the option, and why it has been "hidden" was because people overvalued it. The removal was aimed precisely at people that didn't want to see "Elite" because they didn't consider themselves that good, so they dropped it to good. However, "Good" in the skill preset should be more like "Average" or even lower, because the method of "scaling" the DPS in function of skill poorly represent what happens in reality. 
If a GM wants to know how well someone will perform, they should most definitely use the Elite preset and some other fight styles, because single simulations will never represent well overall results of players. With that information he/she is able to judge how close or far from the best situation the raider is. Comparing it to a bot that can randomly just do garbage won't give him/her good results.

**Comment by N1gh7h4wk** [comment]
I don't see the issue to talk about this more, the option skill= still exists, its just misleading to show it in the GUI so it has been removed from casual accessibility.
